smoke_test/steps/base_step.rb

```diff
@@ -0,0 +1,41 @@
+module SmokeTest
+  module Steps
+    class BaseStep
+      include Capybara::DSL
+
+      attr_accessor :state
+
+      def initialize(state)
+        puts "Step:#{step_name}"
+        @state = state
+      end
+
+      def validate!
+        true
+      end
+
+      def complete_step
+        fail NotImplementedError
```

Comment:
I have a feeling that inheriting from a base class is probably an anti-pattern, and depending on method definitions especially so. I think including shared behaviour via a module leads to better results. In this case, all the base step really gives us is the `with_retries` method (which could be a module on its own). Printing the current step might be more at home in the top-level smoke test, and initialising with `state` would come for free with a `Struct`.